common/lib/plugins/failover/reader_failover_handler.ts

```diff
@@ -322,6 +322,13 @@ class ConnectionAttemptTask {
     );
     try {
       this.targetClient = await this.pluginService.forceConnect(this.newHost, copy);
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        await this.pluginService.forceRefreshHostList();
+        if ((await this.pluginService.getHostRole(this.targetClient)) === HostRole.READER) {
```

Comment:
Wondering if this can be used as the while condition? Also, would it help to enforce a bit of a wait time in the loop?